### PR TITLE
chore: kubernetes-1.11.3 is not a module

### PR DESCRIPTION
**Pull request by pmuir**
Signed-off-by: Pete Muir <pmuir@bleepbleep.org.uk>

#### Submitter checklist

- [ ] Change is code complete and matches issue description.
- [ ] Change is covered by existing or new tests.

#### Which issue this PR fixes

fixes #

### CODE REVIEW

File: Makefile.codegen [hunk 1]
```diff
@@ -26,13 +26,6 @@ generate: build-codegen install-generate-deps generate-mocks generate-openapi ge
 
 install-generate-deps:
 	$(GO) get github.com/petergtz/pegomock/...@$(PEGOMOCK_VERSION)
-	$(GO) get k8s.io/code-generator/cmd/client-gen@$(CLIENTSET_GENERATOR_VERSION)
```
Comment (hferentschik) [inline Makefile.codegen]: Right, technically not needed here, but I had left it to be more explicit

File: cmd/codegen/app/generate_options.go [hunk 2]
```diff
@@ -22,6 +23,7 @@ const (
 	optionBoilerplateFile = "boilerplate-file"
 	optionModuleName      = "module-name"
 	global                = "global"
+	optionVerbose         = "verbose"
```
Comment (hferentschik) [inline cmd/codegen/app/generate_options.go]: 👍

File: cmd/codegen/generator/client_set.go [hunk 3]
```diff
@@ -32,8 +32,8 @@ func InstallCodeGenerators(version string, gopath string) error {
 	}
 	for _, generator := range allGenerators {
 		path := fmt.Sprintf("%s/%s", basePath, generator)
-		util.AppLogger().Infof("installing %s in version %s via 'go get'", path, version)
-		err := util.GoGet(path, version, gopath, true, false)
+		util.AppLogger().Infof("installing %s version %s into %s", path, version, gopath)
+		err := util.GoGet(path, version, gopath, false, false)
```
Comment (hferentschik) [inline cmd/codegen/generator/client_set.go]: Hmm, I don't think this is going to work, is it? With Go modules enabled, I don't think to specify a version works. I guess that is why the generated sources differ now.

File: cmd/codegen/util/commands.go [hunk 4]
```diff
@@ -168,6 +168,8 @@ func (c *Command) RunWithoutRetry() (string, error) {
 	var r string
 	var e error
 
+	AppLogger().Debugf("Running %s %s %s", JoinMap(c.Env, " ", "="), c.Name, strings.Join(c.Args, " "))
```
Comment (hferentschik) [inline cmd/codegen/util/commands.go]: Depending on what you are after you can print a map via `%+v`

**Comment by hferentschik** [comment]
Seems there are some updates to generated files not checked in. Not sure whether the right generator versions are used.

File: cmd/codegen/util/commands.go (hunk 4 above)
Comment (pmuir) [reply cmd/codegen/util/commands.go]: Wanted to print the map as though it were environment variables.

**Comment by pmuir** [comment]
yes - not sure quite what's going on there - i'll check

**Comment by pmuir** [comment]
Ok, correct files are now checked in.

File: cmd/codegen/generator/client_set.go (hunk 3 above)
Comment (pmuir) [reply cmd/codegen/generator/client_set.go]: Right, with Go modules enabled choosing a version is a bit dodgy due to the way go implemented modues - it does all sorts of weird stuff including mucking with your `go.mod`. If you use `go get` without go modules, the code will do the `go get -d` (i.e. download only), pause, adjust the committish in use manually using git, and then run `go install` to complete the `go get`.

In essence `go get` is not really suitable for use in modules mode in the same way as it was before modules - this is quite well documented in the go issue tracker.

Comment (hferentschik) [reply cmd/codegen/generator/client_set.go]: there is `gobin` which bypasses these issues and is probably the way to go for now until the go maintainers introduce better tooling.

Comment (hferentschik) [reply cmd/codegen/generator/client_set.go]: >  If you use go get without go modules, the code will do the go get -d (i.e. download only), pause, adjust the committish in use manually using git, and then run go install to complete the go get.

So, you  are saying it does honor versions, just in an "awkward" way?

**Comment by hferentschik** [comment]
needs a rebase

**Pull request merged**